cmd/testflag.go

```diff
@@ -1,59 +1,175 @@
 package cmd
 
 import (
-	"flag"
+	"errors"
+	"fmt"
+	"strconv"
 	"strings"
 )
 
-// TestFlags appends "-test." to flags used with the test plugin to be passed
-// to the test binary.
-func TestFlags(flags *flag.FlagSet, testArgs []string) []string {
+// testFlagSpec defines a flag we know about.
+type testFlagSpec struct {
+	boolVar    bool // True if the flag is type bool
+	passToTest bool // pass to Test
+	passToAll  bool // pass to test plugin and test binary
+	present    bool // The flag has been seen
+}
+
+// testFlagDefn is the set of flags we process.
+var testFlagDefn = map[string]*testFlagSpec{
```

Comment:
Nah, that's fine, that make sense, please leave it as it is.

On Tue, May 26, 2015 at 7:41 AM, Jesus Alvarez <notifications@github.com>
wrote:

> In cmd/testflag.go
> <https://github.com/constabulary/gb/pull/124#discussion_r30995676>:
>
> >  	"strings"
> >  )
> >
> > -// TestFlags appends "-test." to flags used with the test plugin to be passed
> > -// to the test binary.
> > -func TestFlags(flags *flag.FlagSet, testArgs []string) []string {
> > +// testFlagSpec defines a flag we know about.
> > +type testFlagSpec struct {
> > +	boolVar    bool // True if the flag is type bool
> > +	passToTest bool // pass to Test
> > +	passToAll  bool // pass to test plugin and test binary
> > +	present    bool // The flag has been seen
> > +}
> > +
> > +// testFlagDefn is the set of flags we process.
> > +var testFlagDefn = map[string]*testFlagSpec{
>
> I tried using a value type, but with Go, you can't directly set a struct
> variable like testFlagDefn["flag"].present = true unless it is a
> reference type (see golang/go#3117
> <https://github.com/golang/go/issues/3117>). I can change this behavior
> to use the workaround described in the issue if you wish, or change it to
> something else. FWIW, upstream uses the same technique:
> https://github.com/golang/go/blob/master/src/cmd/go/testflag.go#L249
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/constabulary/gb/pull/124/files#r30995676>.
>